Source/RESTyard.ContractFirst/RESTyard.Generator/Templates/csharp-base/Constructor.razor

```diff
@@ -0,0 +1,120 @@
+    {
+        @:this.@property.name = @Uncapitalize(@property.name);
+    }
+
+    foreach (var operation in Document.Operations)
+    {
+        @:this.@operation.name = @Uncapitalize(@operation.name);
+    }
+
+    foreach (var entity in Document.Entities)
+    {
+        @:this.Entities.AddRange("@entity.collectionName", @Uncapitalize(@entity.collectionName));
+    }
+
+    foreach (var link in Document.Links)
+    {
+        string linkOptionMatch = "";
+        string linkValue = "";
+        if (string.IsNullOrEmpty(link.document))
+        {
+            linkOptionMatch = Uncapitalize(link.rel);
+            linkValue = link.mandatory ? Uncapitalize(link.rel) : "some";
+        }
+        else
+        {
+            var hasQuery = !string.IsNullOrEmpty(link.query);
+            var linkDocument = Schema.Documents.First(d => d.name == link.document);
+            var hasKey = HasKeyProperties(linkDocument);
+            var reference = "HypermediaObject" + (hasQuery ? "Query" : "Key") + "Reference";
+            var type = $"typeof({link.document}Hto)";
+            var tupleParameter = $"{Uncapitalize(link.rel)}Reference";
+            var queryParameter = $"{Uncapitalize(link.rel)}Query";
+            var keyParameter = $"{Uncapitalize(link.rel)}Key";
+            if (!link.mandatory)
+            {
+                if (hasQuery && hasKey)
+                {
+                    linkOptionMatch = tupleParameter;
+                    queryParameter = "some.Query";
+                    keyParameter = "some.Key";
+                }
+                else if (hasQuery)
+                {
+                    linkOptionMatch = queryParameter;
+                    queryParameter = "some";
+                }
+                else if (hasKey)
+                {
+                    linkOptionMatch = keyParameter;
+                    keyParameter = "some";
+                }
+            }
+            else
+            {
+                if (hasQuery && hasKey)
+                {
+                    queryParameter = $"{tupleParameter}.Query";
+                    keyParameter = $"{tupleParameter}.Key";
+                }
+            }
+
+            var parameters = "";
+            if (hasQuery && hasKey)
+            {
+                parameters = $"{queryParameter}, {keyParameter}";
+            }
+            else if (hasQuery)
+            {
+                parameters = queryParameter;
+            }
+            else if (hasKey)
+            {
+                parameters = keyParameter;
+            }
+            else
+            {
+                parameters = "null";
+            }
+
+            linkValue = $"new {reference}({type}, {parameters})";
+        }
+
+        if (link.mandatory)
+        {
+            @:this.Links.Add("@link.rel", @linkValue);
+        }
+        else
+        {
+            @:@linkOptionMatch@(".")Match(some => this.Links.Add("@link.rel", @linkValue));
+        }
+    }
+    @:}
+}
```

Comment:
Is it necessary to have so much code inside the view component? I'd rather have it inside the `@code` block if possible and only the necessary rendering code inside the view. I think it's harder to see what the intended structure of the generated code should be.